src/CheckIfDead.php

```diff
@@ -522,18 +525,22 @@ public function sanitizeURL( $url, $stripFragment = false, $preserveQueryEncodin
 	public function parseURL( $url ) {
 		// Feeding fully encoded URLs will not work.  So let's detect and decode if needed first.
 		// This is just idiot proofing.
-		// First let's break the fragment out to prevent accidentally mistaking a decoded %23 as a #
-		$fragment = parse_url( $url, PHP_URL_FRAGMENT );
-		if ( !is_null( $fragment ) ) {
-			$url = strstr( $url, "#", true );
-		}
-		// Decode URL
-		$url = rawurldecode( $url );
-		// Re-encode the remaining #'s
-		$url = str_replace( "#", "%23", $url );
-		// Reattach the fragment
-		if ( !is_null( $fragment ) ) {
-			$url .= "#$fragment";
+		// Make sure URL is fully encoded by checking if the :// is encoded.
```

Comment:
This comment is confusing. It makes it sound like we expect all URLs passed to parseURL() to be URL-encoded. I imagine you mean to say something like "See if the URL is fully encoded by checking if the :// is encoded."